### PR TITLE
Set selected id to rendered inprogress lines and polygons

### PR DESCRIPTION
**Pull request by drewbo**
Naive approach to #595

### CODE REVIEW

**Comment by mcwhittemore** [comment]
I think this makes sense. Can you add a test to make sure the feature is unselected when someone exits the drawing mode? I'm pretty sure this should happen automatically, but I'd like a test to confirm.

**Comment by drewbo** [comment]
Thanks @mcwhittemore; I had to have the test wait for `afterNextRender` since the selected updating happens in the mode `render` function. Let me know if that is unwieldy at all and needs refactoring

File: test/draw_polygon.test.js [hunk 1]
```diff
@@ -699,3 +699,25 @@ test('draw_polygon touch interaction', t => {
...
+test('draw_polygon is selected while in progress', t => {
+  const container = document.createElement('div');
+  document.body.appendChild(container);
+  const map = createMap({ container });
+  const Draw = new MapboxDraw();
+  map.addControl(Draw);
+  const afterNextRender = setupAfterNextRender(map);
+
+  map.on('load', () => {
+    Draw.changeMode('draw_line_string');
```
Comment (mcwhittemore) [inline test/draw_polygon.test.js]: This should be `draw_polygon`

Comment (drewbo) [inline test/draw_polygon.test.js]: Good call, this is what I get for copy and pasting

**Comment by drewbo** [comment]
I didn't add anything to points since once it is placed, it returns to `'simple_select'` with the point selected (not affected by my changes)

**Comment by mcwhittemore** [comment]
This is great. Can you rebase and then we'll merge.

**Comment by drewbo** [comment]
Rebased

**Comment by mcwhittemore** [comment]
Sorry, @drewbo I meant to rebase this with master so it was up to date. I can do this if the conflicts are a problem.

**Comment by drewbo** [comment]
Ah, my bad. Updated (and I guess squashed too)

**Comment by drewbo** [comment]
@mcwhittemore does this (or should it) emit a `draw.selectionchange` event?

**Comment by mcwhittemore** [comment]
1. Entering a drawing mode should emit a `draw.selectionchange` if something was selected before.
2. Before a shape is valid, `draw.selectionchange` should not be emitted.
3. Once a shape is valid, ... I'm not sure, I think this is up for debate. /cc @davidtheclark 
4. Leaving a draw mode to simple_select
  - with a new shape should result in `draw.selectionchange` if the event **was not** fired in `3`
  - without a new shape should result in `draw.selectionchange` if the event **was** fired in `3`

**Comment by drewbo** [comment]
@mcwhittemore this should be good to go then (it doesn't make any changes to the current events model)

**Comment by mcwhittemore** [comment]
@drewbo the `draw.selectionchange` event shouldn't be emitted until the `Polygon` is valid. This is working of `LineString`.

**Comment by drewbo** [comment]
@mcwhittemore updated the `Polygon` logic to not be selected (and thus not emit the event) until after the 3rd coordinate is drawn (even if it isn't fully added yet)

**Comment by mcwhittemore** [comment]
This is looking good.

We need to update the docs fo [draw.selectionchange](https://github.com/mapbox/mapbox-gl-draw/blob/master/API.md#drawselectionchange)

**Comment by mcwhittemore** [comment]
I'm just noticing that this is a breaking change. :(

**Comment by mcwhittemore** [comment]
I'd like to see this worked out as a custom mode. One hope I have for custom modes, is that we can use them to beta test functionality changes and, if one custom modes gains large popularity, we can switch up how Draw does this.

[Please look here for info on writing custom modes](https://github.com/mapbox/mapbox-gl-draw/blob/master/src/modes/README.md#creating-modes-for-mapbox-draw).

**Pull request closed without merging**